projects/erigon_riscv_proof_sourcing.md

```diff
@@ -0,0 +1,148 @@
+
+The main missing piece is the transpilation module that needs to be written over the next weeks.
+
+### First iteration of the transpilation module (Weeks 5-10)
+The first few weeks will involve us writing the first iteration of the transpilation module. This involves handling the instructions mapping and handling the first version of ELF binary generation.
+
+**Delivery:** We should after this be able to prove simple contracts that we have written ourself, we will also still be in control of the environment. There might be no support for 256bit arithmetic operations, some limits to control flow needs or similar simplifications.
+
+### Stabilizing the transpilation module (Weeks 11-17)
+For the next milestone we want to stabilize and solve some of the [possible challenges](#possible-challenges) to improve our support of real world contracts.
+
+**Delivery:** We should be able to transpile Uniswap V2 and other [real world applications](https://etherscan.io/topstat#HotContracts) and operate in close to a real world environment.
+
+### Optimize, benchmarking and evaluation (Weeks 18-21)
+At this point we hope to have an implementation that we can run experiments on to get a sense of the feasibility of this approach. We want to benchmark this against other existing approaches. The result of this should be published as part of the conclusion of this project.
+
+**Delivery**: We should provide a quantitative performance comparison between our transpilation approach and existing alternatives including metrics such as transpilation time, binary size, and execution speed.
+
+## Possible challenges
+This is a big project and there will obviously be some challenges along the way, we list some of them with possible mitigations that we will try to implement.
+
+### ISA edge case mapping
+- **Challenge**: Different ISAs handle edge cases differently (numeric overflows, memory modeling, instruction errors)
+- **Impact**: Critical - security issues from incorrect transpilation
+- **Mitigation**: Test suite comparing outputs between environments
+- **Timeline**: Ongoing throughout the project
+
+### Transpilation correctness
+- **Challenge**: Bugs in transpilation module pose serious security risks
+- **Impact**: Critical - incorrect translation could compromise system security
+- **Mitigation**: Testing, consider [formal verification methods](https://verified-zkevm.org/)
+- **Timeline**: Continuous verification as opcodes are implemented
+
+### Word size
+- **Challenge**: RISC-V uses 32/64-bit words, EVM uses 256-bit
+- **Impact**: High - affects all arithmetic operations
+- **Mitigation**: There are existing algorithms for supporting arbitrary-precision arithmetic which we can implement in RISC-V.
```

Comment:
Makes sense, I just wonder what performance impact this will have. 256 vs 32/64 is still a big difference, a single ADD opcode can transpile into dozens of RISCV instructions.

Reply:
That is a good point, reworded the sentence a bit and also highlighted that we can use vector processing instructions to possibly migrate this 🤞 

9878d096a8233783092390317d9d87db4f541c1e